apps/system/utils/security_level_cmd.c

```diff
@@ -0,0 +1,128 @@
+
+#include <tinyara/ss_info.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <apps/shell/tash.h>
+#include <security/security_ss.h>
+
+/********************************************************************************
+ * Private Functions
+ ********************************************************************************/
+static void tash_set_security_level(int argc, char **args);
+static void tash_get_security_level(int argc, char **args);
+
+const static tash_cmdlist_t security_level_cmds[] = {
+	{"set_sec_level", tash_set_security_level, TASH_EXECMD_SYNC},
+	{"get_sec_level", tash_get_security_level, TASH_EXECMD_SYNC},
+	{NULL, NULL, 0}
+};
+
+static void show_usage(void) 
+{
+	printf("\nUsage : set_sec_level [options]\n");
+	printf("Set your security level high:\n");
+	printf("	set_sec_level high\n");
+	printf("Set your security level low:\n");
+	printf("	set_sec_level low\n");
+}
+
+static void tash_set_security_level(int argc, char **args)
+{
+	char ss_path[7] = "";
+        security_handle hnd;
+	security_data ss_data = {NULL, 0};
+	security_error ret;
+
+	/* Change security_level according to user input */
+	ss_data.length = 2;
+	if (!strncmp(args[1], "high", 5)) {
+		ss_data.data = "1";
+	} else if (!strncmp(args[1], "low", 4)) {
+		ss_data.data = "0";
+	} else if (!strncmp(args[1], "--help", 7)) {
+		show_usage();
+		return;
+	} else {
+		printf("Invalid argument\n");
+		printf("Use [--help] option if you want to know how to use this command\n");
+		return;
+	}
+
+        /* init */
+	ret = security_init(&hnd);
+        if (ret != SECURITY_OK) {
+		printf("[security_init] error ret : %d\n", ret);
+		return;
+	}
+
+	/* Convert secure storage slot index to string path */
+	snprintf(ss_path, 7, "ss/%d", SECURITY_LEVEL_SS_SLOT_INDEX);
+
+        /* Write 'security_level' variable to secure storage  */
+	ret = ss_write_secure_storage(hnd, ss_path, 0, &ss_data);
+        if (ret != SECURITY_OK) {
+		printf("[ss_write_secure_storage] error ret : %d\n", ret);
+		return;
+	}
+
+	printf("Security level is set\n");
+}
+
+static void tash_get_security_level(int argc, char **args)
+{
+	char ss_path[7] = "";
+        security_handle hnd;
+	security_data ss_data = {NULL, 0};
+	security_error ret;
+
+        /* init */
+	ret = security_init(&hnd);
+        if (ret != SECURITY_OK) {
+		printf("[security_init] error ret : %d\n", ret);
+		return;
+	}
+
+	/* Convert secure storage slot index to string path */
+	snprintf(ss_path, 7, "ss/%d", SECURITY_LEVEL_SS_SLOT_INDEX);
+
+	/* Read 'security_level' variable from secure storage  */
+	ret = ss_read_secure_storage(hnd, ss_path, 0, &ss_data);
+        if (ret != SECURITY_OK) {
+		printf("[ss_read_secure_storage] error ret : %d\n", ret);
+		return;
+	}
+
+	if (!strncmp(ss_data.data, "1", 2)) {
+		printf("security level : high\n");
+	} else if (!strncmp(ss_data.data, "0", 2)) {
+		printf("security level : low\n");
+	} else {
+		printf("Failed to get security_level\n");
```

Comment:
Let's add read value at error sentence. If you print just failure, we can't debug the issue.